### ISSUE TITLE
Allow users to change the font used in the editor

### ISSUE THREAD

**Issue by hannahkane** [mozilla/thimble.mozilla.org#1599]
From a community member: "Is there any way to swap the fonts on the editor? I have a student with a visual impairment that makes it very hard for her to read sans serif fonts."

**Comment by ryanwarsaw** [comment]
if I could get some UX insight on the implementation of this, I'd be more than glad to take this ticket.

**Comment by flukeout** [comment]
@hannahkane Do you know if there is a specific font they want used? I'm guessing it's a system they already have on their machine? Is it enough to offer some options (like from Google fonts?)

**Comment by hannahkane** [comment]
I asked but don't believe I've heard back yet. I think offering a few options (including a serif one) would be good.

**Comment by flukeout** [comment]
Hey @ryanwarsaw - I think dropping this setting into the editor gear menu would work pretty well.

Maybe a dropdown below Text Size...

![image](https://cloud.githubusercontent.com/assets/25212/21999610/da6fb7ae-dbef-11e6-9650-7065221cbba0.png)

Alternatively, we could just have two options, serif and sans-serif and have a toggle between the two. We'd probably want to carry the changes through to the file-tree as well.

@hannahkane let us know if you get any more details. Also, is there an existing font that the user finds the easiest to read?

**Comment by flukeout** [comment]
Some reading here http://www.textmatters.com/resources/pdfs/visImpd_typogTM.pdf

Some takeaways

* Font face less important than size, contrast, spacing & weight
* Best weight is medium/semibold
* Avoid _italics_
* Make sure the numbers are easy to read and distinguish from letters
* Text in all caps is bad (EDITOR / PREVIEW) etc.. !
* More space between lines is good
* Shorter line length may be preferable
  * Users can set this by changing the width of the editor in our case

**Comment by ryanwarsaw** [comment]
@flukeout I'd agree that the editor menu seems like the most sensible place for this. It'd be interesting to explore possibly providing a few font options, but also allowing people to use their own. Perhaps even assigning font/text related options to their Thimble account, so it persists across projects?

I'm not knowledgeable on visual impairments but from reading your summary above, something like Office Code Pro (https://github.com/nathco/Office-Code-Pro) might be worth looking into?

**Comment by hannahkane** [comment]
I asked, but haven't yet heard back from the person who originally requested this, so I think we should just follow best practices.

**Comment by flukeout** [comment]
Yeah, I'm curious because the original request seems like a specific case or ailment. The general guidelines don't mention serifs as being more legible.

Maybe we should hold off until we can get more clarity from the original request?

**Comment by hannahkane** [comment]
Asking the original requester again now...

**Comment by hannahkane** [comment]
From the original requester: "Another teacher in the building said as soon as she started printing her classwork in Times New Roman instead of Arial, the student with the disability started producing much higher quality work. 

So I think any Serif font would work. If I had to pick one then Times would be it since my colleague has confirmed that it is readable."

**Comment by gideonthomas** [comment]
Just removing the "assigned to contributor" flag for now since I can assign @ryanwarsaw to the issue instead.

**Comment by flukeout** [comment]
Any progress on this @ryanwarsaw - ? I've recently bumped the font weight in the entire UI as well, so hopefully that helps a bit.

**Comment by ryanwarsaw** [comment]
@flukeout I prioritized working on the passport JS implementation, and helping contributors setup their Thimble environments over this, I'd like to land a fix in the near future, but isn't on my radar at the moment.

**Comment by ryanwarsaw** [comment]
This ticket is available for anyone who is interested in fixing it, I don't think I'm going to be able to get to this for at least the next couple of weeks, so I'm un-assigning myself.
